lib/zoho_resource_client.rb

```diff
@@ -0,0 +1,73 @@
+class ZohoResourceClient
+  ZOHO_ORG_ID = ArchiveConfig.ZOHO_ORG_ID
+  CONTACT_SEARCH_ENDPOINT = 'https://desk.zoho.com/api/v1/contacts/search'.freeze
+  CONTACT_CREATE_ENDPOINT = 'https://desk.zoho.com/api/v1/contacts'.freeze
+  TICKET_CREATE_ENDPOINT = 'https://desk.zoho.com/api/v1/tickets'.freeze
+
+  def initialize(access_token:, email:, username: nil)
+    @access_token = access_token
+    @email = email
+    @username = username
+  end
+
+  def retrieve_contact_id
+    (find_contact || create_contact).fetch('id')
+  end
+
+  def create_ticket(ticket_attributes:)
+    response_raw = HTTParty.post(
+      TICKET_CREATE_ENDPOINT,
+      headers: headers,
+      body: ticket_attributes.to_json
+    )
+    JSON.parse(response_raw.body)
+  end
+
+  private
+
+  def find_contact
+    response_raw = HTTParty.get(
+      CONTACT_SEARCH_ENDPOINT + '?' + query_string(search_params),
+      headers: headers,
+    )
+    return if response_raw.nil?
+
+    JSON.parse(response_raw.body).fetch('data').first
+  end
+
+  def create_contact
+    response_raw = HTTParty.post(
+      CONTACT_CREATE_ENDPOINT,
+      headers: headers,
+      body: contact_body.to_json
+    )
+    JSON.parse(response_raw.body)
+  end
+
+  def search_params
```

Comment:
Should we have a comment that says despite us explicitly limiting it to 1 result, there should only ever be one client per email address because Zoho requires emails to be unique? Just because it was a pain in the neck to find that information (assuming it's accurate, lolsob).

Also, is there any reason we might want to call this `contact_search_params`? Or could the same method potentially hold search parameters for other things as well?

Reply:
We're not searching for anything else yet, was my main reason. It could be reused to search for anything that holds an email field, yes.